### ISSUE TITLE
Cross-Platform Math API

### ISSUE THREAD

**Issue by maluoi** [maluoi/StereoKit#90]
[DirectXMath](https://github.com/microsoft/DirectXMath) is the current math solution, and it's good for now, but has limited long-term appeal as it doesn't support targets like WASM. A great alternative would be [RealtimeMath](https://github.com/nfrechette/rtm)

**Comment by vpenades** [comment]
In case of C#, what the library should use is System.Numerics.Vectors, which is the de facto standard types for Vectors and matrices.

I understand there might be times when some internal structs are needed to interop with C, but in the end, the types provided by the StereoKit APi closely match the ones of System.Numerics.Vectors.

So, I don't know if it's possible to do this replacement anymore... but at the very least, StereoKit vector/matrix types should provide implicit conversions from system.numerics.vectors types, so people can use them seamlessly.

**Comment by maluoi** [comment]
Having come from Unity and XNA before that, I actually had no idea C# added Vector types until just a few months ago! Since discovering that, I definitely do agree StereoKit should use them. I've been adding some implicit conversions as holdovers, but I do need to check interop and come up with a more solid plan for moving that over :)

Thanks for bringing it up! I'll create a separate work item for this.

**Comment by vpenades** [comment]
Yes, it's amazing it took so long to microsoft to add a vector math library to the platform. In fact, it was lifted from the XNA library almost entirely.

**Comment by maluoi** [comment]
I knew XNA lived on in the DirectXTK, but didn't realize it was also the foundation for C#'s Numerics implementation too! That's pretty sweet :)

**Comment by maluoi** [comment]
Dropping an implementation note in here regarding DirectXMath, it looks like [Emscripten _does_ support SIMD](https://emscripten.org/docs/porting/simd.html), so there's a chance the native side may work without changes in WASM after all! Will have to test this out, see if it compiles or if there's any major catches.

**Comment by vpenades** [comment]
Keep in mind that the c# system.numerics.vectors also supports SIMD...

**Comment by maluoi** [comment]
This issue was meant to be primarily about the native side of StereoKit's math. In particular, cross-platform compatibility! The stuff I've got so far works everywhere I'm targeting except for web :) DirectXMath doesn't have a non-SIMD backend, which was the holdup before I discovered Emscripten can do SIMD stuff with the right flags.

The C# side is a separate issue in my mind, which is why I branched that off into #97. Right now StereoKit C# dips into the native side for math so it uses SIMD features, but I haven't checked yet how marshaling affects performance there. One reason why I'd prefer to use Numerics is because I won't have to worry about the effects of marshaling for math at all. It would also take care of library interop, which makes it extra appealing. I recognize those are separate features though, so if it turns out Numerics doesn't play well with my native code, I'd still absolutely want to provide compatibility code for Numerics regardless!
